File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.js
function solution(arr, divisor) {
    let answer = [], cnt = 0;
    arr.sort((a,b)=> a-b);
    for(let i of arr){
        if(i%divisor === 0) answer.push(i)
        else cnt++
    }
    if(cnt === arr.length) return[-1]
    return answer;
}

File: Java/프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public ArrayList<Integer> solution(int[] arr, int divisor) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        Arrays.sort(arr);
        for(int num : arr) {
            if(num %divisor == 0) answer.add(num);
        }
        
        if(answer.size() == 0) answer.add(-1);
            
        return answer;
    }
}